src/VisualStudio/Core/Impl/CodeModel/NodeKeyValidation.cs

```diff
@@ -17,14 +17,17 @@ public NodeKeyValidation()
 
         public void AddProject(AbstractProject project)
         {
-            var provider = (IProjectCodeModelProvider)project;
-            var fcms = provider.ProjectCodeModel.GetCachedFileCodeModelInstances();
-
-            foreach (var fcm in fcms)
+            var provider = project as IProjectCodeModelProvider;
+            if (provider != null)
```

Comment:
Ideally we could possibly do this without needing to directly access these specific classes.  @jasonmalinowski might have ideas on where we can go here in the future.  But this was by far the lowest cost way to be able to integrate with valuable infrastructure pieces like the VisualStudioProjectTracker.    Without that, we'd likely have had to write thousands of lines of glue code.  And that makes :panda_face: :sob: